components/core/src/DictionaryWriter.hpp

```diff
@@ -53,16 +53,9 @@ class DictionaryWriter {
     void close ();
 
     /**
-     * Gets the entry with the given ID
-     * @param id
-     * @return Pointer to the entry with the given ID
+     * Writes dict_type count and flush compressors to file
```

Comment:
I try to leave implementation details out of the header comment (if necessary, those should be in the function body or in the header comment of the function implementation). So here, I wouldn't mention `dict_type count` or the compressors; instead, how about something like `// Writes the dictionary's header and flushes any unwritten content to disk`?